### ISSUE TITLE
404 instead of desired page since 0.2.0

### ISSUE THREAD

**Issue by visualcookie** [wiziple/gatsby-plugin-intl#27]
Hey.

Since version`0.2.0`, I get the 404 page instead of the e.g. index pages.

**Comment by soft4net** [comment]
Hello, I experience the same problem after updating to gatsby 2.9.0. I'm using gatsby-plugin-intl version 0.2.4.

**Comment by visualcookie** [comment]
> Hello, I experience the same problem after updating to gatsby 2.9.0. I'm using gatsby-plugin-intl version 0.2.4.

Ah, thanks for the headsup. Could also be Gatsby in this case.

**Comment by soft4net** [comment]
I also downgraded to gatsby 2.8.8, before run develop or build I executed:
$ gatsby clean

And for me routing start working again.
I will try with gatsby 2.9.1 which was released.

**Comment by visualcookie** [comment]
I used the example now and it works again. I don't know what I did wrong tbh, since I basically used the same setup. The only thing I see is, that the Gatsby version the example is using is `2.8.4` and I had `2.8.6` installed.

**Comment by rock345** [comment]
We are also seeing this issue.  We downgraded the plugin to get past it.

**Comment by wiziple** [comment]
Yeah I thought I made some mistakes since I released a new version yesterday.
So I spent quite time debugging the plugin itself and I had to bump the version 3 times.🤣

This issue can be temporarily fixed by installing specific gatsby version under 2.9.0.
`npm install gatsby@2.8.6 —save`

I will continue to figure out what causes the issue and update here.

**Comment by rock345** [comment]
Thanks for the workaround!

**Comment by rchrdnsh** [comment]
having the same issue on gatsby 2.8.2.

**Comment by rchrdnsh** [comment]
also having the same problem with 2.8.6.

**Comment by rchrdnsh** [comment]
also having the same issue with 2.8.6 of gatsby and 0.2.4 of this plugin

**Comment by rchrdnsh** [comment]
I can't seem to find any combination of versions of anything that end up working. I'm completely dead in the water in this current project now...

**Comment by visualcookie** [comment]
@rchrdnsh I just used the example project inside the plugin repo for now and it works like a charm. Still don't know what's causing the issue.

**Comment by rchrdnsh** [comment]
the example project works for me too, but not the project i'm actually working on...no idea why, either :-(

**Comment by rchrdnsh** [comment]
ok, yay! I had to:

`npm install gatsby@2.8.6 --save`

3 different times before it would work, but now the app is working again!

sooooooo strange...

**Comment by rchrdnsh** [comment]
sigh...

back to my original issue, which is that the site now works again on my local machine, but the routes are broken when I push and deploy on Netlify...

www.alma.vacations

**Comment by gvocale** [comment]
Same here. Issue with gatsby 2.9.2. Downgrading gatsby to 2.8~ fixes it. I'll follow up this thread for updates.

**Comment by anicolaides** [comment]
Heads up, same occurs when upgrading from ```"browser-lang": "0.0.1"``` to ```"browser-lang": "0.1.0"```.

**Comment by wiziple** [comment]
This is caused by 2.9.0 changes. 
https://github.com/gatsbyjs/gatsby/issues/14742

**Comment by jmyatt** [comment]
we pinned gatsby and the plugin as follows 9 days ago which worked fine and now hitting the same issue without having changed this:

```
"gatsby": "2.8.6",
"gatsby-plugin-intl": "0.1.7",
```

**Comment by lewislbr** [comment]
any news in this front? thanks

**Comment by rchrdnsh** [comment]
the gatsby team is almost done with a fix, they are just working through some tests, I think.

**Comment by rchrdnsh** [comment]
This one:

https://github.com/gatsbyjs/gatsby/pull/15010

**Comment by sidharthachatterjee** [comment]
Fixed and published in `gatsby@2.10.4`

Sorry for the trouble!

**Comment by rchrdnsh** [comment]
Hi @sidharthachatterjee! 😁

Thank you and your team for all your hard work on this issue...

buuuuuut...

it's still not quite working for me...

I ran `gatsby info`:

```bash
System:
    OS: macOS 10.14.5
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.12.0 - /usr/local/bin/node
    Yarn: 1.10.1 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 75.0.3770.100
    Firefox: 67.0
    Safari: 12.1.1
  npmPackages:
    gatsby: ^2.10.4 => 2.10.4
    gatsby-cli: ^2.7.2 => 2.7.2
    gatsby-image: ^2.2.3 => 2.2.3
    gatsby-mdx: ^0.6.3 => 0.6.3
    gatsby-plugin-intl: ^0.2.4 => 0.2.4
    gatsby-plugin-layout: ^1.1.0 => 1.1.0
    gatsby-plugin-manifest: ^2.2.0 => 2.2.0
    gatsby-plugin-netlify: ^2.1.0 => 2.1.0
    gatsby-plugin-offline: ^2.2.0 => 2.2.0
    gatsby-plugin-react-helmet: ^3.1.0 => 3.1.0
    gatsby-plugin-sharp: ^2.2.1 => 2.2.1
    gatsby-plugin-styled-components: ^3.1.0 => 3.1.0
    gatsby-plugin-twitter: ^2.1.0 => 2.1.0
    gatsby-plugin-web-font-loader: ^1.0.4 => 1.0.4
    gatsby-source-filesystem: ^2.1.0 => 2.1.0
    gatsby-source-instagram: ^0.5.0 => 0.5.0
    gatsby-transformer-remark: ^2.5.0 => 2.5.0
    gatsby-transformer-sharp: ^2.2.0 => 2.2.0
    gatsby-transformer-yaml: ^2.2.0 => 2.2.0
  npmGlobalPackages:
    gatsby-cli: 2.7.0
```

...and deleted the cache, package-lock and npm modules, then re-installed all of them again, then ran the site, and the problem of the router not working for me is still persisting.

If you have a moment, maybe you can take a look at the repo and see if you can spot anything that I might still be doing wrong?

https://github.com/arrowgtp/alma

Also, here is the URL of the latest build pushed to Netlify:

www.alma.vacations

**Comment by wiziple** [comment]
https://github.com/gatsbyjs/gatsby/issues/15101

Hello, I submitted this issue with more explanation, so the let's wait 👍

**Comment by wiziple** [comment]
A workaround fix published on `gatsby-plugin-intl@0.2.5`
This is a temporary fix until the gatsby team resolves the matchPath issue.

**Comment by giacomoalonzi** [comment]
I incurred in the same issue, you can see the problem [here](https://covidlive-it-git-feature-multilang.bardaxx.now.sh/en/)

` "gatsby": "^2.19.49",`
`"gatsby-plugin-intl": "^0.3.3",`

any hint?
